File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-py/models/product.py
from __future__ import annotations

from pydantic import BaseModel


class Product(BaseModel):
    productId: int
    supplierId: int
    name: str
    description: str | None = None
    price: float
    sku: str
    unit: str
    imgName: str | None = None
    discount: float


class CreateProductRequest(BaseModel):
    supplierId: int
    name: str
    description: str | None = None
    price: float
    sku: str
    unit: str
    imgName: str | None = None
    discount: float = 0.0


class UpdateProductRequest(BaseModel):
    supplierId: int | None = None
    name: str | None = None
    description: str | None = None
    price: float | None = None
    sku: str | None = None
    unit: str | None = None
    imgName: str | None = None
    discount: float | None = None

File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-cs/Models/Product.cs
namespace OctoSupply.Api.Models;

public sealed record Product(
    int ProductId,
    int SupplierId,
    string Name,
    string? Description,
    double Price,
    string Sku,
    string Unit,
    string? ImgName,
    double Discount);

public sealed class CreateProductRequest
{
    public int SupplierId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public double Price { get; set; }
    public string Sku { get; set; } = string.Empty;
    public string Unit { get; set; } = string.Empty;
    public string? ImgName { get; set; }
    public double Discount { get; set; }
}

public sealed class UpdateProductRequest
{
    public int? SupplierId { get; set; }
    public string? Name { get; set; }
    public string? Description { get; set; }
    public double? Price { get; set; }
    public string? Sku { get; set; }
    public string? Unit { get; set; }
    public string? ImgName { get; set; }
    public double? Discount { get; set; }
}
